schemas/2015-08-01/Microsoft.Compute.json

```diff
@@ -1,549 +1,835 @@
+            }
+          }
+        },
+        "type": "object",
+        "required": [
+          "networkInterfaceConfigurations"
+        ]
+      },
+      "virtualMachineProfile": {
+        "properties": {
+          "osProfile": {
+            "$ref": "#/definitions/virtualMachineScaleSetOsProfile"
+          },
+          "storageProfile": {
+            "$ref": "#/definitions/virtualMachineScaleSetStorageProfile"
+          },
+          "extensionProfile": {
+            "$ref": "#/definitions/virtualMachineScaleSetExtensionProfile"
+          },
+          "networkProfile": {
+            "$ref": "#/definitions/virtualMachineScaleSetNetworkProfile"
+          }
+        },
+        "type": "object",
+        "required": [
+          "osProfile",
+          "storageProfile",
+          "extensionProfile",
+          "networkProfile"
+        ]
+      },
+      "virtualMachineScaleSets": {
+        "type": "object",
+        "properties": {
+          "type": {
+            "enum": [
+              "Microsoft.Compute/virtualMachineScaleSets"
+            ]
+          },
+          "apiVersion": {
+            "enum": [
+              "2015-05-01-preview",
+              "2015-06-15"
+            ]
+          },
+          "properties": {
+            "sku": {
+              "$ref": "#/definitions/sku"
+            },
             "properties": {
-                "name": {
-                    "type": "string"
-                },
-                "diskSizeGB": {
-                    "type": "string"
-                },
-                "lun": {
-                    "type": "number"
-                },
-                "vhd": {
-                    "$ref": "#/definitions/vhdUri"
-                },
-                "caching": {
-                    "type": "string"
-                },
-                "createOption": {
-                    "type": "string"
-                }
+              "upgradePolicy": {
+                "$ref": "#/definitions/upgradePolicy"
+              },
+              "virtualMachineProfile": {
+                "$ref": "#/definitions/virtualMachineProfile"
+              }
             },
             "type": "object",
             "required": [
-                "name",
-                "lun",
-                "vhd",
-                "createOption"
+              "upgradePolicy",
+              "virtualMachineProfile"
             ]
-        },
-        "storageProfile": {
-            "properties": {
-                "imageReference": {
-                    "$ref": "#/definitions/imageReference"
-                },
-                "osDisk": {
-                    "$ref": "#/definitions/osDisk"
-                },
-                "dataDisks": {
-                    "type": "array",
-                    "items": {
-                        "$ref": "#/definitions/dataDisk"
+          },
+          "resources": {
+            "type": "array",
+            "items": {
+              "allOf": [
```

Comment:
If all the possible child resources are these 3, you can remove the "resources" part, all resource types can have these child resources by default, if they are added into "2015-01-01\deploymentTemplate.json" Ln 40 and "2014-04-01-preview\deploymentTemplate.json" Ln 43.